### PR TITLE
Leonep2

### PR DESCRIPTION
**Pull request by pleone55**
implemented encryption and decryption of email to determine whether the user exists and if they have already taken the specific quizzes given to them. Fixed time limit on quiz countdown. Added checks and modal to determine if all questions were answered and if time limit hit 0. Updated route that handles sending the quiz link to contain the quiz id. Respondent info is prefilled if user exists and cannot update their info.

### CODE REVIEW

File: client/src/components/pages/RespondentQuiz.js [hunk 1]
```diff
@@ -7,27 +7,31 @@ import {
...
+    const [confirm, setConfirm] = useState(false);
 
     let time = quiz_resp && quiz_resp.timeLimit;
-    const [stateTimeLimit, setStateTimeLimit] = useState(0);
+    const [stateTimeLimit, setStateTimeLimit] = useState(1);
 
 	const onTimeLimitChange = useCallback(() => {
 		setInterval(
-			() => setStateTimeLimit((stateTimeLimit) => stateTimeLimit - 1),
+			() => setStateTimeLimit((stateTimeLimit) => stateTimeLimit - 34),
```
Comment (pamelayin) [inline client/src/components/pages/RespondentQuiz.js:28]: do you mind explaining what this means?

File: client/src/components/pages/RespondentQuiz.js [hunk 2]
```diff
@@ -7,27 +7,31 @@ import {
...
     const navigate = useNavigate();
 
     const [questionsAnswered, setQuestionsAnswered] = useState([]);
+    const [alert, setShowAlert] = useState(false);
+    const [confirmMessage, setConfirmMessage] = useState('');
+    const [confirm, setConfirm] = useState(false);
 
     let time = quiz_resp && quiz_resp.timeLimit;
-    const [stateTimeLimit, setStateTimeLimit] = useState(0);
+    const [stateTimeLimit, setStateTimeLimit] = useState(1);
```
Comment (pamelayin) [inline client/src/components/pages/RespondentQuiz.js:24]: good catch, found this while I was testing

File: client/src/components/pages/RespondentQuiz.js (hunk 1 above)
Comment (pleone55) [reply client/src/components/pages/RespondentQuiz.js:28]: Oops this needs to be reverted back to just - 1. I did that for some of the tests for quicker results and I forgot to change it back. Let me push the edit

**Review by pamelayin — APPROVED** [review]
looks great. thanks for getting the hash out of the way. I found that you have added the {iv} to the link to respondent quiz. I'm not too clear on the connection with the hashkey of the user. Just for my info, do you mind going through the mechanism how it works with me? 

Otherwise, I just had one comment on the timer of quiz which I added on the respodentquiz.js

**Pull request merged**